goodwe/sensor.py

```diff
@@ -52,22 +55,21 @@ def power_unit(self):
 
     def decode_power(self, value: int) -> int:
         """Decode human readable value of power parameter"""
-        if self == ScheduleType.ECO_MODE:
-            return value
-        elif self == ScheduleType.PEAK_SHAVING:
+        if self == ScheduleType.PEAK_SHAVING:
             return value * 10
-        if self == ScheduleType.ECO_MODE_745:
+        elif self == ScheduleType.ECO_MODE_745:
             return int(value / 10)
+        elif self == ScheduleType.NOT_SET:
+            if value < -100 or value > 100:
+                return int(value / 10)
         else:
             return value
 
-    def encode_power(self, value: int) -> int:
+    def encode_power(self, eco_mode, value: int) -> int:
         """Encode human readable value of power parameter"""
-        if self == ScheduleType.ECO_MODE:
-            return value
-        elif self == ScheduleType.PEAK_SHAVING:
+        if self == ScheduleType.PEAK_SHAVING:
             return int(value / 10)
-        if self == ScheduleType.ECO_MODE_745:
+        elif eco_mode == ScheduleType.ECO_MODE_745:
```

Comment:
We cannot use self when encoding power since self is the previous mode and not the new mode. Will probably not work for Peak shaving either since we are not in peak shaving mode yet when encoding. I have no solution for peak shaving so I left it as is.